modules/asset-registry/src/lib.rs

```diff
@@ -275,6 +280,105 @@ impl<T: Config> ForeignAssetIdMapping<ForeignAssetId, MultiLocation, AssetMetada
 	}
 }
 
+/// Simple fee calculator that requires payment in a single fungible at a fixed rate.
+///
+/// The constant `FixedRate` type parameter should be the concrete fungible ID and the amount of it
+/// required for one second of weight.
+pub struct FixedRateOfForeignAsset<T, FixedRate: Get<u128>, R: TakeRevenue> {
```

Comment:
Is the generic param `T` used somewhere?

Reply:
https://github.com/AcalaNetwork/Acala/blob/21dfd9edca4012dd730147ca32227fb019709531/modules/asset-registry/src/lib.rs#L316

used here.

Reply:
the rate will be different for each asset. it should read minimal balance of the asset and use ratio between native token ED to determine the weight per second